src/test/java/com/thomashan/coup/turn/StandardTurnTest.java

```diff
@@ -0,0 +1,208 @@
+
+        assertEquals(nextPlayer, turn.getPlayer());
+    }
+
+    @Test
+    public void testGetActionHistory_GivenInitialTurn_ReturnsEmptyActionHistory() {
+        assertIterableEquals(Collections.emptyList(), turn.getActionHistory());
+    }
+
+    @Test
+    public void testGetActionHistory_GivenActionPerformed_ReturnsCorrectActionHistory() {
+        Action action = MainAction.of(player, INCOME);
+        turn = turn.perform(action);
+
+        assertIterableEquals(Collections.singletonList(action), turn.getActionHistory());
+    }
+
+    @Test
+    public void testGetAllowableActions_GivenInitialTurn_ReturnsCorrectActions() {
+        Player player2 = players.get(1);
+        Player player3 = players.get(2);
+        List<Action> actions = turn.getAllowableActions();
+
+        assertIterableEquals(Arrays.asList(
+                MainAction.of(player, TAX),
+                MainAction.of(player, STEAL, player2),
+                MainAction.of(player, STEAL, player3),
+                MainAction.of(player, EXCHANGE),
+                MainAction.of(player, INCOME),
+                MainAction.of(player, FOREIGN_AID)
+        ), turn.getAllowableActions());
+    }
+
+    @Test
+    public void testGetAllowableActions_GivenWaitingChallengeMainActionState_ReturnsCorrectActions() {
+        Player player2 = players.get(1);
+        Player player3 = players.get(2);
+        turn = turn.perform(MainAction.of(player, STEAL, player2));
+
+        assertIterableEquals(Arrays.asList(
+                ChallengeAction.of(player2, ALLOW),
+                ChallengeAction.of(player2, CHALLENGE),
+                ChallengeAction.of(player3, ALLOW),
+                ChallengeAction.of(player3, CHALLENGE)
+        ), turn.getAllowableActions());
+    }
+
+    @Test
+    public void testGetAllowableActions_GivenWaitingBlockActionState_ReturnsTarget() {
+        Player player2 = players.get(1);
+        Player player3 = players.get(2);
+        turn = turn.perform(MainAction.of(player, STEAL, player2))
+                .perform(ChallengeAction.of(player2, ALLOW))
+                .perform(ChallengeAction.of(player3, ALLOW));
+
+        assertIterableEquals(Arrays.asList(
+                BlockAction.block(player2, BLOCK_STEAL, CAPTAIN),
+                BlockAction.block(player2, BLOCK_STEAL, AMBASSADOR),
+                BlockAction.noBlock(player2)
+        ), turn.getAllowableActions());
+    }
+
+    @Test
+    public void testGetAllowableActions_GivenWaitingChallengeBlockActionState_ReturnsCorrectActions() {
+        Player player2 = players.get(1);
+        Player player3 = players.get(2);
+        turn = turn.perform(MainAction.of(player, STEAL, player2))
+                .perform(ChallengeAction.of(player2, ALLOW))
+                .perform(ChallengeAction.of(player3, ALLOW))
+                .perform(BlockAction.block(player2, BLOCK_STEAL, CAPTAIN));
+
+        List<Action> allowableActions = turn.getAllowableActions();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Found 'DU'-anomaly for variable 'allowableActions' (lines '171'-'179').](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4561385)